File: ios/PortalApiSmokeTest.swift
import Foundation

// MARK: - Smoke test helper (manual use only)
// Not invoked by default; keep as a reference snippet.

@available(iOS 15.0, *)
func portalApiSmokeTest() async {
    let api = PortalApiClient(
        config: .init(baseUrl: "http://127.0.0.1:5001/monsoonfire-portal/us-central1")
    )

    let payload = CreateBatchRequest(
        ownerUid: "USER_UID",
        ownerDisplayName: "Display Name",
        title: "Test batch",
        kilnName: nil,
        intakeMode: "STAFF_HANDOFF",
        estimatedCostCents: 2500,
        estimateNotes: nil,
        notes: nil
    )

    do {
        _ = try await api.createBatch(
            idToken: "ID_TOKEN",
            adminToken: "ADMIN_TOKEN",
            payload: payload
        )
    } catch {
        print("portalApiSmokeTest failed: \(error)")
    }
}

File: android/app/src/main/java/com/monsoonfire/portal/reference/PortalApiSmokeTest.kt
package com.monsoonfire.portal.reference

// Manual smoke test helper (reference only)

fun portalApiSmokeTest(idToken: String, adminToken: String) {
    val api = PortalApiClient(PortalApiClient.Config(baseUrl = "http://127.0.0.1:5001/monsoonfire-portal/us-central1"))

    val payload = CreateBatchRequest(
        ownerUid = "USER_UID",
        ownerDisplayName = "Display Name",
        title = "Test batch",
        kilnName = null,
        intakeMode = "STAFF_HANDOFF",
        estimatedCostCents = 2500,
        estimateNotes = null,
        notes = null
    )

    try {
        api.createBatch(
            idToken = idToken,
            adminToken = adminToken,
            payload = payload
        )
    } catch (err: PortalApiException) {
        println("portalApiSmokeTest failed: ${err.message}")
        println(err.meta)
    }
}
